pkg/protocol/protocol.go

```diff
@@ -232,7 +231,11 @@ func (p *Protocol) initChainManager() {
 	}, event.WithWorkerPool(wp))
 
 	p.Events.Engine.SlotGadget.SlotFinalized.Hook(func(index iotago.SlotIndex) {
-		rootCommitment := p.MainEngineInstance().EarliestRootCommitment()
+		rootCommitment, valid := p.MainEngineInstance().EarliestRootCommitment(index)
+		fmt.Println("Slot finalized:", index, "root commitment:", rootCommitment, "valid:", valid)
```

Comment:
It would be nice to actually have a properly formatted log for this! All the logging is a separate topic altogether though.

Reply:
I agree but I think we should come up with a general solution soon. I'll leave that print for now and we should replace it in a follow up PR.